stdlib/public/runtime/CMakeLists.txt

```diff
@@ -82,26 +82,37 @@ add_swift_library(swiftRuntime OBJECT_LIBRARY TARGET_LIBRARY
   C_COMPILE_FLAGS ${swift_runtime_compile_flags}
   INSTALL_IN_COMPONENT never_install)
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux" OR CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
-  add_swift_library(section_magic_begin STATIC IS_STDLIB IS_STDLIB_CORE
+set(ELFISH_SDKS)
+foreach(sdk ${SWIFT_CONFIGURED_SDKS})
+  is_elfish_sdk("${sdk}" IS_EFLISH)
+  if(IS_ELFISH)
+    list(APPEND EFLISH_SDKS "${sdk}")
+  endif()
+endforeach()
+
+add_swift_library(section_magic_begin OBJECT_LIBRARY IS_STDLIB IS_STDLIB_CORE
     swift_sections.S
     C_COMPILE_FLAGS ${swift_runtime_compile_flags} "-DSWIFT_BEGIN"
+    TARGET_SDKS "${ELFISH_SDKS}"
     INSTALL_IN_COMPONENT never_install)
-  add_swift_library(section_magic_end STATIC IS_STDLIB IS_STDLIB_CORE
+add_swift_library(section_magic_end OBJECT_LIBRARY IS_STDLIB IS_STDLIB_CORE
     swift_sections.S
     C_COMPILE_FLAGS ${swift_runtime_compile_flags} "-DSWIFT_END"
+    TARGET_SDKS "${ELFISH_SDKS}"
     INSTALL_IN_COMPONENT never_install)
-endif()
 
 set(object_target_list)
 foreach(sdk ${SWIFT_CONFIGURED_SDKS})
-  if("${sdk}" STREQUAL "LINUX" OR "${sdk}" STREQUAL "FREEBSD" OR "${sdk}" STREQUAL "ANDROID")
+  is_elfish_sdk("${sdk}" IS_ELFISH)
+  if(IS_ELFISH)
```

Comment:
A random suggestion, but adding `${SWIFT_SDK_${sdk}_OBJECT_FORMAT}` would help to write these conditions more compactly:

```cmake
if("${SWIFT_SDK_${sdk}_OBJECT_FORMAT}" STREQUAL "ELF")
```